Update docs to cover uv managed installations (#2072)
plone/documentation: docs/admin-guide/override-core.md

Plone includes a few hundred Python packages.
Sometimes you will need to override one or more package versions to fix a bug.

## Override the version of a core Plone package

> Choose a tab depending on your Python package manager.

Reviewer: ```suggestion
For instructions of how to override a core Plone package, select the tab below according to your Python package manager.
```

`````{tab-set}

````{tab-item} uv

```{tip}

> Use the following instructions if you have a project that was created using Cookieplone after March 2026.

Reviewer: If I create a project today, that's during March, and I'd have no instructions to follow.

An explicit date is preferred. Better yet, let's refer to Cookieplone v1.0.0 instead of a date. In which case, this would be a `versionchanged` admonition instead of `tip`.

If necessary, we can also use nested tabs for pre-1.0.0 and 1.0.0+.

Reviewer: Actually, now that I see it applies for both uv and pip, let's move this admonition before the tabbed UI so that readers know which is relevant to their situation before they click a tab.

Author: We can't use the Cookieplone version. The change happened in cookieplone-templates, not Cookieplone. Cookieplone clones the main branch of cookieplone-templates (unless it's told to use a specific tag by setting the COOKIEPLONE_REPOSITORY_TAG environment variable, or some other set of templates). There's actually no release of Cookieplone 2.x yet, but the updated templates are already used.

That's why I went with something based on a date. I can update it to mention a specific day. And I like the idea to move the admonition above the tabs to help the reader decide. Will update.

Author: Updated in https://github.com/plone/documentation/pull/2072/commits/cd84028519885bb0ee59dc3a8f0a66a2c7d37948

Author: Actually I realized using a date doesn't work well either, because the classic_project template hasn't been updated to a uv-managed approach. I updated it to refer to the managed setting in pyproject.toml as a way to tell which way to use.

Reviewer: See https://github.com/plone/documentation/pull/2072#issuecomment-4132204551.

Author: I don't see how your suggestion there helps, since we have to explain how to check if the project is uv-managed regardless of the cookieplone-templates version.

I did have an idea for a simpler way to check this, which I'll explain in a new comment on the current version of the file.

{doc}`run-plone`
```

````

````{tab-item} pip

```{tip}
Use the following instructions if you have a project that was created using Cookieplone before March 2026.

> ```

Reviewer: See above comment.

Add a version override to the file {file}`mx.ini`.
This example uses `plone.api`.

```
[settings]
version-overrides =
    plone.api==2.0.0a3
```

```{seealso}
The {file}`mx.ini` file configures a tool called {term}`mxdev`.

> For an explanation of why Plone uses `mxdev`, see {ref}`manage-packages-mxdev-label`.

Reviewer: ```suggestion
For an explanation of why Plone uses `mxdev` for projects using pip as the Python package manager, see {ref}`manage-packages-mxdev-label`.
```

{doc}`run-plone`
```

````

````{tab-item} Buildout

```{tip}
Use the following instructions if you installed Plone with Buildout.

> ```

Reviewer: This is redundant to the instruction "Choose a tab depending on your Python package manager."

```suggestion
```

Author: I was thinking it would provide confirmation to the user that they're on the right tab, but I trust your judgment here.

{doc}`run-plone`
```

````

`````

## Install a core Plone package from source

A core Plone package can be installed from a source control system such as GitHub.

> Choose a tab depending on your Python package manager.

Reviewer: From here onward, I'd like to make sure that the instructions for selecting a given tab are consistent with my suggestions above.

`````{tab-set}

````{tab-item} uv

```{tip}
Use the following instructions if you have a project that was created using Cookieplone after March 2026.
```

This example uses `plone.restapi`.

Clone the repository into a local directory.

```shell
git clone git@github.com:plone/plone.restapi.git
```

> Add the local directory to your `uv` project as an editable package.

Reviewer: The name is just plain old "uv", and not an inline literal. If it's a command, then `uv` is appropriate.

```suggestion
Add the local directory to your uv project as an editable package.
```
